components/logged_in/logged_in.tsx

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Redirect} from 'react-router';
+import {UserProfile} from 'mattermost-redux/src/types/users';
+import {viewChannel} from 'mattermost-redux/actions/channels';
+import semver from 'semver';
+import {Channel} from 'mattermost-redux/types/channels';
+
+import * as GlobalActions from 'actions/global_actions.jsx';
+import * as WebSocketActions from 'actions/websocket_actions.jsx';
+import * as UserAgent from 'utils/user_agent';
+import LoadingScreen from 'components/loading_screen';
+import {getBrowserTimezone} from 'utils/timezone.jsx';
+import store from 'stores/redux_store.jsx';
+import WebSocketClient from 'client/web_websocket_client.jsx';
+import BrowserStore from 'stores/browser_store';
+
+
+const dispatch = store.dispatch;
+const getState = store.getState;
+
+const BACKSPACE_CHAR = 8;
+
+type Props = {
+    currentUser?: UserProfile | Record<string, unknown> | null,
```

Comment:
This should just be `currentUser?: UserProfile` if possible. I don't think `getUserProfile` can return null, and we ideally shouldn't support arbitrary objects here. If you run into issues because the tests are trying to pass an empty object, it'd be better to have the test use `{} as UserProfile` so that the non-test code will be more strictly typed.